client_test/integration_test.go

```diff
@@ -0,0 +1,178 @@
+package client_test
+
+import (
+	"math/big"
+	"os"
+	"testing"
+
+	"github.com/ethereum/go-ethereum/common"
+	"github.com/statechannels/go-nitro/client"
+	"github.com/statechannels/go-nitro/client/query"
+	"github.com/statechannels/go-nitro/internal/testactors"
+	td "github.com/statechannels/go-nitro/internal/testdata"
+	"github.com/statechannels/go-nitro/protocols"
+	"github.com/statechannels/go-nitro/types"
+)
+
+var simpleCase = TestCase{
+	Description:    "Simple test",
+	Chain:          MockChain,
+	MessageService: TestMessageService,
+	NumOfChannels:  1,
+	MessageDelay:   0,
+	LogName:        "simple_integration_run.log",
+	NumOfHops:      1,
+	NumOfPayments:  1,
+	Participants: []TestParticipant{
+		{StoreType: MemStore, Name: testactors.AliceName},
+		{StoreType: MemStore, Name: testactors.BobName},
+		{StoreType: MemStore, Name: testactors.IreneName},
+	},
+}
+
+var complexCase = TestCase{
+	Description:    "Complex test",
+	Chain:          SimulatedChain,
+	MessageService: P2PMessageService,
+	NumOfChannels:  5,
+	MessageDelay:   0,
+	LogName:        "complex_integration_run.log",
+	NumOfHops:      2,
+	NumOfPayments:  5,
+	Participants: []TestParticipant{
+		{StoreType: DurableStore, Name: testactors.AliceName},
+		{StoreType: DurableStore, Name: testactors.BobName},
+		{StoreType: DurableStore, Name: testactors.IreneName},
+		{StoreType: DurableStore, Name: testactors.BrianName},
+	},
+}
+
+var cases = []TestCase{simpleCase, complexCase}
+
+func TestClientIntegration(t *testing.T) {
+	// Clean up all the test data we create at the end of the test
+	defer os.RemoveAll(STORE_TEST_DATA_FOLDER)
+
+	for _, tc := range cases {
+		t.Run(tc.Description, func(t *testing.T) {
+			err := tc.Validate()
+			if err != nil {
+				t.Fatal(err)
+			}
+			infra := setupSharedInra(tc)
+
+			// Setup clients
+			clientA := setupIntegrationClient(tc, testactors.AliceName, infra)
+			clientB := setupIntegrationClient(tc, testactors.BobName, infra)
+
+			intermediaries := []client.Client{setupIntegrationClient(tc, testactors.IreneName, infra)}
+			intermediaryAddresses := []types.Address{*intermediaries[0].Address}
+			if tc.NumOfHops == 2 {
+				intermediaries = append(intermediaries, setupIntegrationClient(tc, testactors.BrianName, infra))
+				intermediaryAddresses = append(intermediaryAddresses, *intermediaries[1].Address)
+			}
+
+			defer clientA.Close()
```

Comment:
It might be clearer to defer client close calls immediately below each client instantiation. In other words:
```
clientA := ...
defer clientA.Close()
```

Reply:
Updated in https://github.com/statechannels/go-nitro/pull/1193/commits/f2f6009fe5eacd03af2c2c97797b45614a78da75